crates/astria-sequencer-relayer/tests/blackbox/main.rs

```diff
@@ -1,90 +1,432 @@
 #![allow(clippy::missing_panics_doc)]
 
-pub mod helper;
+pub mod helpers;
 
-use std::{
-    collections::HashSet,
-    time::Duration,
-};
+use std::collections::HashSet;
 
-use assert_json_diff::assert_json_include;
-use helper::{
-    CometBftBlockToMount,
+use astria_core::{
+    primitive::v1::RollupId,
+    protocol::test_utils::ConfigureSequencerBlock,
+};
+use helpers::{
+    SequencerBlockToMount,
     TestSequencerRelayerConfig,
 };
 use reqwest::StatusCode;
-use serde_json::json;
-use tokio::time::{
-    sleep,
-    timeout,
-};
+use tendermint::account::Id as AccountId;
 
+const RELAY_SELF: bool = true;
 const RELAY_ALL: bool = false;
 
+#[tokio::test(flavor = "multi_thread", worker_threads = 1)]
+async fn one_block_is_relayed_to_celestia() {
+    let sequencer_relayer = TestSequencerRelayerConfig {
+        relay_only_self: false,
+        last_written_sequencer_height: None,
+        only_include_rollups: HashSet::new(),
+    }
+    .spawn_relayer()
+    .await;
+
+    let _guard = sequencer_relayer.mount_abci_response(1).await;
+    let block_to_mount = SequencerBlockToMount::GoodAtHeight(1);
+    let _guard = sequencer_relayer
+        .mount_sequencer_block_response::<RELAY_ALL>(block_to_mount, "good block 1")
+        .await;
+    let _guard = sequencer_relayer
+        .mount_celestia_app_broadcast_tx_response("broadcast tx 1")
+        .await;
+    let get_tx_guard = sequencer_relayer
+        .mount_celestia_app_get_tx_response(53, "get tx 1")
+        .await;
+    // The `MIN_POLL_INTERVAL_SECS` is 1, meaning the relayer waits for 1 second before attempting
+    // the first `GetTx`, so we wait for 2 seconds.
+    sequencer_relayer
+        .timeout_ms(
+            2_000,
+            "waiting for get tx guard",
+            get_tx_guard.wait_until_satisfied(),
+        )
+        .await;
+
+    // Assert the relayer reports the correct Celestia and sequencer heights.
+    sequencer_relayer
+        .wait_for_latest_confirmed_celestia_height(53, 1_000)
+        .await;
+    sequencer_relayer
+        .wait_for_latest_fetched_sequencer_height(1, 1_000)
+        .await;
+    sequencer_relayer
+        .wait_for_latest_observed_sequencer_height(1, 1_000)
+        .await;
+
+    assert_eq!(
+        sequencer_relayer.celestia_app_received_blob_count(),
```

Comment:
This is very close to the original implementation (the celestia mock server using channels etc), but I think it might be nicer if we expressed this in terms of a mock (or series of mocks) together with `wait_until_satisfied`.

Reply:
Agreed, but as mentioned above, I'll leave for a follow-up PR.